R/as.data.table.R

```diff
@@ -133,6 +133,22 @@ as.data.table.list = function(x,
   missing.check.names = missing(check.names)
   origListNames = if (missing(.named)) names(x) else NULL  # as.data.table called directly, not from inside data.table() which provides .named, #3854
   empty_atomic = FALSE
+
+  #Handle keep.rownames for vectors (mimicking data.frame behavior)
+  vector_rownames = NULL
+  if(!isFALSE(keep.rownames)) {
+    for(i in seq_len(n)){
+      xi = x[[i]]
+      if (!is.null(xi) && is.atomic(xi) && !is.null(names(xi)) && is.null(dim(xi)) && length(names(xi)) > 0) {
```

Comment:
IINM the condition is redundant

```suggestion
      if (!is.null(xi) && is.atomic(xi) && !is.null(names(xi)) && is.null(dim(xi))) {
```